display_list/display_list_path_effect.h

```diff
@@ -0,0 +1,319 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_DISPLAY_LIST_PATH_EFFECT_H_
+#define FLUTTER_DISPLAY_LIST_DISPLAY_LIST_PATH_EFFECT_H_
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include "flutter/display_list/display_list_attributes.h"
+#include "flutter/display_list/types.h"
+#include "flutter/fml/logging.h"
+
+namespace flutter {
+
+class DlComposePathEffect;
+class DlSumPathEffect;
+class DlCornerPathEffect;
+class DlDashPathEffect;
+class DlDiscretePathEffect;
```

Comment:
Where are these constructed from? We should only implement what we need to support Flutter and Impeller.

Flutter's dart:ui doesn't have any path effects so none of them arise from a dependency there.

The text package (SkParagraph) does use dash path effects, making it the only variant of this object that we should support. It may be obsolete soon, though, because Impeller is implementing a new text package and we may implement dashing some other way.

Note that if you scan SkParagraph, it appears to use Compose and Discrete, but if you look more closely you will see that the Discrete path effects are only instantiated with NOP data that causes them to be implemented by nullptr, and Compose will just return the non-null argument it receives in that case. So, only dash is really used by SkParagraph (double check this in case I misread something.)